firestore/src/main/firestore_main.h

```diff
@@ -184,6 +186,7 @@ class FirestoreInternal {
   std::unordered_set<ListenerRegistrationInternal*> listeners_;
 
   std::shared_ptr<util::Executor> transaction_executor_;
+  const char* database_name_;
```

Comment:
When does `database_name_` delete? Because this is a pointer, you need to manage lifecycle yourself. I have traced logic, and it appears the pointer is created by customer, and can be deleted by customer as well during the lifetime of Firestore object. We need to atleast make a copy, but maybe just use `std::string` here instead.

In case I missed something, could you please explain the lifecycle of the `database_name_` pointer?